common/tests/unit/test_CipherEngine.cpp

```diff
@@ -0,0 +1,148 @@
+#define BOOST_TEST_MAIN
+
+#define BOOST_TEST_MODULE keyEncryptionDecryption
+#include <boost/filesystem.hpp>
+#include <boost/test/unit_test.hpp>
+//Local private includes
+#include "../../include/common/CipherEngine.hpp"
+#include "../../include/common/Util.hpp"
+
+#include "common/libjson.hpp"
+
+//Standard includes
+#include <string.h>
+#include <iostream>
+#include <fstream>
+
+using namespace std;
+using namespace SDMS;
+
+BOOST_AUTO_TEST_SUITE(KeyEncryptionDecryptionTest)
+
+BOOST_AUTO_TEST_CASE(test_EncryptionDecryption)
+{
+
+    std::cout << "THIS IS A NEW FLAG" << std::endl;
+    unsigned char key[32];
+    //CipherEngine::generateEncryptionKey(key);
+    readFile("../../build/core/server/datafed-token-key.txt", 32, key);
```

Comment:
magic number

Reply:
Also, would it be better to create the file as part of the test? This has hard coded in the build folder depending on what cmake command a user runs it may not exist 

```
cmake -S. -B different_build_folder
```

Then the build folder won't be called build.